src/targets/__tests__/commitOnGitRepository.test.ts

```diff
@@ -0,0 +1,51 @@
+import { pushArchiveToGitRepository } from '../commitOnGitRepository';
+
+const mockClone = jest.fn();
+const mockCheckout = jest.fn();
+const mockRaw = jest.fn();
+const mockCommit = jest.fn();
+const mockAddTag = jest.fn();
+const mockPushTags = jest.fn();
+
+// eslint-disable-next-line no-var
+var mockExecSync: jest.Mock;
```

Comment:
hmmm does it not work with the same approach here? (so `const mockExecSync = jest.fn()` and then use this in the `jest.mock` call?) this is sometimes weird, but just to be sure - if that does not work maybe add a comment here why this is that way?

Reply:
I think jest hoists the .mock() calls and they end up above the binding and if it's a `const` it will throw. `var`s get hoisted so they're fine. I sometimes hate jest. I'll play around with it a bit more and if I can't find a clean solution I'll just add a comment.